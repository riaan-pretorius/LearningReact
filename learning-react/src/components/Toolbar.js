// @flow
import React, { Component } from "react";
import { Row, Col, Nav } from "reactstrap";
import ToolbarItem from "./ToolbarItem";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

type Props = {
  title: string,
  links: Array<Object>
};
type State = {
  title: string,
  links: Array<Object>
};

class Toolbar extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      title: props.title,
      links: props.links
    };
  }
  getLinks() {
    return this.state.links.map((link, index) => {
      return <ToolbarItem key={index} to={link.to} text={link.text} />;
    });
  }
  render() {
    return (
      <Col>
        <Row>
          <Col sm="4">
            <h3 className="text-muted">{this.state.title}</h3>
          </Col>
          <Col sm="8">
            <Nav style={styles.nav} pills>
              {this.getLinks()}
            </Nav>
          </Col>
        </Row>
      </Col>
    );
  }
}

const styles = {
  nav: {
    float: "right"
  }
};

export default Toolbar;
