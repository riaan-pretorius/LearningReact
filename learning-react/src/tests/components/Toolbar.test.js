import React from 'react'
import Toolbar from '../../components/Toolbar'
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer'
import {
  MemoryRouter
} from 'react-router-dom'

test('Toolbar Test Snapshot', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Toolbar title="Testing" links={[{to: "#", text: "Testing"}]}/>
    </MemoryRouter>
  )
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
})

test('Toolbar Test', () => {
    const component = mount(
      <MemoryRouter>
        <Toolbar title="Testing" links={[{to: "/", text: "Testing"}]}/>
      </MemoryRouter>
    )
    
    let header = component.find('h3')
    expect(header.exists()).toBeTruthy()
    expect(header.text()).toBe("Testing")

    let a = component.find('a')
    expect(a.exists()).toBeTruthy()
    expect(a.text()).toBe("Testing")
    expect(a.prop('href')).toBe("/")
})