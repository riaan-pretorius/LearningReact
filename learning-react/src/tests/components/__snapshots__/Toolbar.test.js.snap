// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Test Snapshot 1`] = `
<div
  className="col"
>
  <div
    className="row"
  >
    <div
      className="col col-sm-4"
    >
      <h3
        className="text-muted"
      >
        Testing
      </h3>
    </div>
    <div
      className="col col-sm-8"
    >
      <ul
        className="nav nav-pills"
        style={
          Object {
            "float": "right",
          }
        }
      >
        <li
          className="nav-item"
        >
          <a
            aria-current={false}
            className="nav-link"
            href="#"
            onClick={[Function]}
            style={undefined}
          >
            Testing
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
